homeassistant/components/sensor/netatmo_public.py

```diff
@@ -82,60 +102,84 @@ def name(self):
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity."""
-        return 'mm'
+        return self._unit_of_measurement
 
     def update(self):
         """Get the latest data from NetAtmo API and updates the states."""
         self.netatmo_data.update()
-        self._state = self.netatmo_data.data
+
+        if self.netatmo_data.data is None:
+            _LOGGER.warning("No data found for %s", self._name)
+            self._state = STATE_UNKNOWN
+            return
+
+        data = None
+
+        if self.type == 'temperature':
+            data = self.netatmo_data.data.getLatestTemperatures()
+        elif self.type == 'pressure':
+            data = self.netatmo_data.data.getLatestPressures()
+        elif self.type == 'humidity':
+            data = self.netatmo_data.data.getLatestHumidities()
+        elif self.type == 'rain':
+            data = self.netatmo_data.data.getLatestRain()
+        elif self.type == 'windstrength':
+            data = self.netatmo_data.data.getLatestWindStrengths()
+        elif self.type == 'guststrength':
+            data = self.netatmo_data.data.getLatestGustStrengths()
+
+        if not data:
+            _LOGGER.warning("No station provides %s data in the area %s",
+                            self.type, self._area_name)
+            self._state = STATE_UNKNOWN
+            return
+
+        if self._mode == 'avg':
+            self._state = round(sum(data.values()) / len(data), 1)
+        elif self._mode == 'max':
+            self._state = max(data.values())
 
 
 class NetatmoPublicData:
     """Get the latest data from NetAtmo."""
 
-    def __init__(self, auth, lat_ne, lon_ne, lat_sw, lon_sw, calculation):
+    def __init__(self, auth, lat_ne, lon_ne, lat_sw, lon_sw):
         """Initialize the data object."""
         self.auth = auth
         self.data = None
         self.lat_ne = lat_ne
         self.lon_ne = lon_ne
         self.lat_sw = lat_sw
         self.lon_sw = lon_sw
-        self.calculation = calculation
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Request an update from the Netatmo API."""
         import pyatmo
-        raindata = pyatmo.PublicData(self.auth,
-                                     LAT_NE=self.lat_ne,
-                                     LON_NE=self.lon_ne,
-                                     LAT_SW=self.lat_sw,
-                                     LON_SW=self.lon_sw,
-                                     required_data_type="rain")
-
-        if raindata.CountStationInArea() == 0:
-            _LOGGER.warning('No Rain Station available in this area.')
+        data = pyatmo.PublicData(self.auth,
+                                 LAT_NE=self.lat_ne,
+                                 LON_NE=self.lon_ne,
+                                 LAT_SW=self.lat_sw,
+                                 LON_SW=self.lon_sw,
+                                 filtering=True)
```

Comment:
What is filtering?

Reply:
Filtering makes the API response exclude abnormal sensor readings. For example if someone installs an outside sensor indoors.